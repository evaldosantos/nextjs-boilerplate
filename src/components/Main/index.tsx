import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <S.Logo src="/img/logo.svg" alt="React Boilerplate Logo" />
    <S.Title>React Boilerplate</S.Title>
    <S.Description>
      TypeScript, ReactJS, NextJS e Styled Components
    </S.Description>
  </S.Wrapper>
)

export default Main
