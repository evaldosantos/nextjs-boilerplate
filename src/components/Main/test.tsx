import { render, screen } from '@testing-library/react'

import Main from './index'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react boilerplate/i })
    ).toBeInTheDocument()

    expect(container).toMatchInlineSnapshot(`
      <div>
        <main>
          <h1>
            React Boilerplate
          </h1>
        </main>
      </div>
    `)
  })
})
