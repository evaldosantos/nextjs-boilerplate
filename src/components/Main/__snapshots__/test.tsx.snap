// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 3rem;
}

.c1 {
  width: 25rem;
}

.c2 {
  font-size: 2.5rem;
  padding: 2rem;
}

<div>
  <main
    class="c0"
  >
    <img
      alt="React Boilerplate Logo"
      class="c1"
      src="/img/logo.svg"
    />
    <h1
      class="c2"
    >
      React Boilerplate
    </h1>
    <h2
      class="styles__Description-sc-19t6oi2-3"
    >
      TypeScript, ReactJS, NextJS e Styled Components
    </h2>
  </main>
</div>
`
